test/integration/cds_integration_test.cc

```diff
@@ -222,6 +222,123 @@ TEST_P(CdsIntegrationTest, CdsClusterTeardownWhileConnecting) {
   EXPECT_LE(cx_counter->value(), 1);
 }
 
+class LazyInitClusterStatsTest : public CdsIntegrationTest {
+
+public:
+  void initialize() {
+    this->use_real_stats_ = true;
+    config_helper_.addConfigModifier([](::envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+      bootstrap.set_enable_lazyinit_stats(true);
+    });
+    CdsIntegrationTest::initialize();
+    test_server_->waitForCounterGe("cluster_manager.cluster_added", 1);
+  }
+
+  void SendRequestToCluster1AndWaitForResponse() {
+    BufferingStreamDecoderPtr response = IntegrationUtil::makeSingleRequest(
+        lookupPort("http"), "GET", "/cluster1", "", downstream_protocol_, version_, "foo.com");
+    ASSERT_TRUE(response->complete());
+    cleanupUpstreamAndDownstream();
+  };
+
+  void ExpectStats(absl::string_view cluster, bool initialized_gauge_exists, int initialized_gauge,
+                   bool cx_active_exists, int upstream_cx_active) {
+    if (initialized_gauge_exists) {
+      EXPECT_EQ(
+          test_server_->gauge(absl::StrCat("cluster.", cluster, ".ClusterTrafficStats.initialized"))
+              ->value(),
+          initialized_gauge);
+    } else {
+      EXPECT_EQ(test_server_->gauge(
+                    absl::StrCat("cluster.", cluster, ".ClusterTrafficStats.initialized")),
+                nullptr);
+    }
+    if (cx_active_exists) {
+      EXPECT_EQ(
+          test_server_->counter(absl::StrCat("cluster.", cluster, ".upstream_cx_total"))->value(),
+          upstream_cx_active);
+    } else {
+      EXPECT_EQ(test_server_->counter(absl::StrCat("cluster.", cluster, ".upstream_cx_total")),
+                nullptr);
+    }
+  }
+  void UpdateCluster1() {
+    envoy::config::cluster::v3::Cluster cluster1_updated = cluster_creator_(
+        ClusterName1, fake_upstreams_[UpstreamIndex2]->localAddress()->ip()->port(),
+        Network::Test::getLoopbackAddressString(ipVersion()), "ROUND_ROBIN");
+    sendDiscoveryResponse<envoy::config::cluster::v3::Cluster>(
```

Comment:
can you make a small helper for this call? It would do 2 things:
 * make the call sites slightly smaller
 * make it easier to see the difference betweeen the call-sites.

This would factor out this long call
```
  sendDiscoveryResponse<envoy::config::cluster::v3::Cluster>(Config::TypeUrl::get().Cluster, ...)
```
and allow us to focus on what's different between the calls.

Reply:
sorry, I am lost, are you suggesting that we add a helper function for this one single line, that's spreaded in this whole file?

Reply:
well it's two lines now because it's so verbose, and all the interesting stuff is harder to find.. Yes that's what I'm suggesting.

I didn't realize that is in the whole file, and I think you should go ahead and put the helper in CdsIntegrationTest and change those call-sites also.

the call sites will look like:
```
   sendClusterDiscoveryResponse({cluster1_updated}, {cluster1_updated}, {}, "42");
```

Reply:
what you are suggesting is essentially specializing the template. 
do you mind if I create a cleanup issue, that apply the thing you suggested globally under test/**? 

$ grep sendDiscoveryResponse -R test/ | wc -l 
216

Reply:
No, just making a simple wrapper for it. This is a pretty trivial change.

Reply:
oh I was not suggesting doing it across all tests; just this one file.

Reply:
I am a little bit leaning towards the consistency side. Let me know if you think differently.

Reply:
Yes that's why I made this suggestion in this file :)

I'm not against consistency but I want to make the tests readable now so I can figure out what's covered and what's missing without reading through so much boilerplate.